README.md

```diff
@@ -59,12 +63,137 @@ $ npm i @mixmaxhq/promise-pool
 
 Changelog
 ---------
+
+* 1.2.0 Add `maxPending` option to avoid problematic usage (see new [Troubleshooting](#troubleshooting) section)
+
 * 1.1.1 Move `ava` and `ava-spec` to `devDependencies`.
 
 * 1.1.0 Adds transpilation so it can be used in Node 6 (and prior) environments.
 
 * 1.0.0 Initial release.
 
+Troubleshooting
+---------------
+
+### `cannot queue function in pool`
+
+If you're getting this error, then one of two things is happening:
```

Comment:
A short blurb about why (by default, you can't have more than one pending request) would be helpful here - it took me until the 4th example to see the reasoning behind it.

Reply:
Good call!

Reply:
f66f280